pyvista/core/filters/data_set.py

```diff
@@ -90,10 +91,10 @@ def _clip_with_function(
 
     def align(
         self,
-        target,
-        max_landmarks=100,
-        max_mean_distance=1e-5,
-        max_iterations=500,
+        target: pyvista.DataSet,
```

Comment:
For this case we can use a simple union `vtkDataSet | DataSet`. We need both otherwise there can be false positive type errors since `DataSet` is not a subclass of `vtkDataSet`. 

I think this is OK for now but it would be nice if we can solve this more elegantly. I ran into a similar typing issue in #5790.

Reply:
Changed in [1f39364](https://github.com/pyvista/pyvista/pull/6828/commits/1f39364deaf58b15cea1daf9ee3b3f9b0d1fa8bc). For some filters that did not accept `vtkDataSet` directly I removed the limitation and wrapped the input. So, all filters which have a second mesh as input should now be typed as `DataSet  | vtkDataSet`

Reply:
> For this case we can use a simple union `vtkDataSet | DataSet`. We need both otherwise there can be false positive type errors since `DataSet` is not a subclass of `vtkDataSet`.
> 
> I think this is OK for now but it would be nice if we can solve this more elegantly. I ran into a similar typing issue in #6790.

A good solution I think is to use `DataSetType` from #6854 instead of `DataSet`.